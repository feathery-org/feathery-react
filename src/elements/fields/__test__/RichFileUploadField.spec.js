import React from 'react';
import { create, act } from 'react-test-renderer';
import Elements from '../..';

describe('RichFileUploadField', () => {
  it('renders a basic file upload component', () => {
    // Arrange
    const props = {
      element: {
        id: 'fieldId',
        styles: {
          height: '50',
          height_unit: '%',
          width: '50',
          width_unit: 'px',
          background_color: '000000',
          uploader_padding_top: '0',
          uploader_padding_bottom: '0',
          uploader_padding_left: '0',
          uploader_padding_right: '0',
          cta_padding_top: '0',
          cta_padding_bottom: '0',
          cta_padding_left: '0',
          cta_padding_right: '0'
        },
        mobile_styles: {},
        servar: {
          key: 'key',
          name: 'Upload file...',
          required: true,
          metadata: {
            file_types: 'image/*'
          }
        },
        properties: {
          required: true,
          metadata: {
            file_types: 'image/*'
          },
          icon_url: 'https://feathery.io'
        }
      },
      required: true,
      onChange: jest.fn(),
      onClick: jest.fn()
    };

    // Act
    let component;
    act(() => {
      component = create(<Elements.RichFileUploadField {...props} />);
    });

    // Assert
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
