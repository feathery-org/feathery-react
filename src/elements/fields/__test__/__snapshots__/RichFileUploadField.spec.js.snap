// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichFileUploadField renders a basic file upload component 1`] = `
Array [
  <div
    className="css-rxpr37-RichFileUploadField"
    id="key"
    onClick={[Function]}
  >
    <img
      className="img-fluid"
      src="https://feathery.tech"
      style={
        Object {
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": 0,
        }
      }
    />
    <div
      className="css-1bv0rem-RichFileUploadField"
    >
      Upload file...
    </div>
    
    
    <input
      accept="image/*"
      onChange={[Function]}
      required={true}
      style={
        Object {
          "bottom": 0,
          "height": 1,
          "opacity": 0,
          "position": "absolute",
          "width": 1,
        }
      }
      type="file"
    />
  </div>,
  "",
]
`;
